### DISCUSSION TITLE
Rolling Train and Test Sets

### DISCUSSION THREAD

**Discussion by tiredofmakingnewusernames** [kogalur/randomForestSRC#346] [Q&A]
Relative newbie to ML with a question. 
(1) My data are a timeseries with a continuous response variable (a rate of gas flux) with several continuous covariates and a few factors. My understanding is that randomForestSRC is, in fact, splitting data into train and test sets behind the scenes, thus it is not necessary to do this myself.  Is it possible to specify in the package to do this on a rolling basis to avoid the problem of "[using the future to predict the past](https://medium.com/@soumyachess1496/cross-validation-in-time-series-566ae4981ce4)"?

**Comment by ishwaran** [comment]
Yes, the package uses out-of-bag (OOB) data to construct an OOB ensemble from which test set performance is obtained without the need to use classical cross-validation, CV.  This OOB ensemble and the test set error can be likened to leave-one-out CV; ie CV(1).

However, with time series data as you point out, we have to be careful and therefore the CV(1) estimator is not appropriate.  Therefore, you will have to manually split the data using "rolling basis" or some other adjustment used for CV with time series.
